tests/rptest/tests/partition_movement_test.py

```diff
@@ -656,7 +656,15 @@ def test_deletion_stops_move(self, num_to_upgrade):
         # check that the status is in progress
 
         def get_status():
-            partition_info = admin.get_partitions(topic, partition)
+            try:
+                partition_info = admin.get_partitions(topic, partition)
+            except requests.exceptions.HTTPError as e:
+                if e.errno == 404:
```

Comment:
looks good to me. i could also see it being reasonable to return `None` from `get_partitions`.